common/str.cpp

```diff
@@ -990,6 +990,21 @@ String toPrintable(const String &in, bool keepNewLines) {
 	return res;
 }
 
+String percentEncodeString(const String &src) {
+	String res;
+
+	for (uint i = 0; i < src.size(); i++) {
+		char c = src[i];
+		if ((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') ||
```

Comment:
You can use utility functions for these checks, found in common/util.h. e.g. check (`isAlNum()`)[https://github.com/scummvm/scummvm/blob/83b3d3a91bd2267e4601aee62e1d244f9a571f6d/common/util.h#L267]